### PR TITLE
fix: REST deployment issue due to missing ecs:TagResource permission

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

Fixes https://github.com/aws-amplify/amplify-category-api/issues/2299 of missing ecs:TagResource permission during REST API deployment for API plugin Gen-1 customers. This is identical to the [fix in ECS's CDK construct](https://github.com/aws/aws-cdk/pull/29401/files#diff-df545f4b38d5b880dbaeb12b64f532e8f6bf6e88e48501bc88181f5ab8a3f1b2R1013).

#### Description of how you validated changes
Added unit test. Existing container E2Es that test the new customer workflows pass.
[First E2E run - all container related tests pass](https://tiny.amazon.com/x9gatg72/IsenLink)
[Second run for 2 failures](https://tiny.amazon.com/tibv8skc/IsenLink)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sundersc — APPROVED** [review]

File: packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts [hunk 1]
```diff
@@ -192,6 +192,43 @@ export class PipelineWithAwaiter extends Construct {
...
     );
 
+    const ecsDeployActionRole = new iam.Role(scope, 'EcsDeployActionRole', {
+      assumedBy: new iam.AccountRootPrincipal(),
+    });
+    ecsDeployActionRole.addToPolicy(
+      new iam.PolicyStatement({
+        actions: ['ecs:TagResource'],
+        effect: iam.Effect.ALLOW,
+        resources: ['*'],
```
[the remark this answers is not in the document]
Comment (palpatim) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts:202]: Never mind--I see this is the resource that the ECS CDK construct adds for TagResource as well.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts [hunk 2]
```diff
@@ -192,6 +192,43 @@ export class PipelineWithAwaiter extends Construct {
       } as Record<string, codebuild.BuildEnvironmentVariable>,
     );
 
+    const ecsDeployActionRole = new iam.Role(scope, 'EcsDeployActionRole', {
+      assumedBy: new iam.AccountRootPrincipal(),
+    });
+    ecsDeployActionRole.addToPolicy(
```
Comment (palpatim) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts:198]: How does this policy pick up the other required permissions (`ecs:DescribeServices`, `ecs:DescribeTaskDefinition`, etc) when you're only explicitly declaring `ecs:TagResource`?

File: packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts [hunk 3]
```diff
@@ -192,6 +192,43 @@ export class PipelineWithAwaiter extends Construct {
       } as Record<string, codebuild.BuildEnvironmentVariable>,
     );
 
+    const ecsDeployActionRole = new iam.Role(scope, 'EcsDeployActionRole', {
```
Comment (palpatim) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts:195]: Please add a TODO to remove this if/when we are able to upgrade to CDK >= 2.132.0, and add a corresponding acceptance criterion to our CDK upgrade backlog item

File: packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts (hunk 2 above)
Comment (phani-srikar) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts:198]: When we pass this custom role into the CDK construct's `EcsDeployAction`, it **adds** these additional policy statements that it requires to the role's policy document.

**Review by sundersc — APPROVED** [review]

File: packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts (hunk 2 above)
Comment (palpatim) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts:200]: Thanks, saw that in the CDK source that it's adding the policy statements to the incoming role

**Review by palpatim — APPROVED** [review]

**Pull request merged**